File: step1_start_project/src/main/java/migration/simple/config/DBConfiguration.java
package migration.simple.config;

import lombok.*;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConfigurationProperties()
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DBConfiguration {
    private DbConfig db;

    @Bean
    public DbConfig configureDb() {
        return new DbConfig(db.getUrl(), db.getUser(), unSecure(db.getPassword()));
    }

    private String unSecure(String password) {
        int secretIndex = password.indexOf("Secret");

        return password.substring(0, secretIndex);
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class DbConfig {
        private String url;
        private String user;
        private String password;
    }
}

File: step4_migration_to_webflux/src/main/kotlin/migration/simple/config/DBConfiguration.kt
package migration.simple.config

import org.springframework.context.annotation.Bean

open class DBConfiguration {
    var db: DbConfig = DbConfig()

    @Bean
    open fun configureDb(): DbConfig {
        return DbConfig(db.url, db.user, unSecure(db.password))
    }

    private fun unSecure(password: String): String {
        val secretIndex = password.indexOf("Secret")

        return password.substring(0, secretIndex)
    }

    data class DbConfig(
            var url: String = "",
            var user: String = "",
            var password: String = ""
    )
}
